src/nameutil.rs

```diff
@@ -125,6 +125,28 @@ pub fn lib_name_to_toml(name: &str) -> String {
     name.to_string().replace("-", "_").replace(".", "_")
 }
 
+pub fn shared_lib_name_to_link_name(name: &str) -> String {
+    let mut s = name;
+
+    if s.starts_with("lib") {
+        s = &s[3..];
+    }
+
+    let offset = s.find(".so").unwrap_or(s.len());
+    s = &s[..offset];
+
+    s.to_string()
+}
+
+pub fn shared_libs_to_links(shared_libs: &Vec<String>) -> String {
+    let mut links = vec![];
+    for it in shared_libs {
+        links.push(format!("\"{}\"", shared_lib_name_to_link_name(it)));
+    }
+
+    format!("[{}]", links.join(", "))
```

Comment:
Maybe only if `!links.is_empty()`?

Reply:
oh, empty list, good point :) but just curios it's possible case apart from programming error. (no native library is required for binding?)

Reply:
That's a good point. Should probably directly error out then instead

Reply:
now it will cause panic in that case

Reply:
Not nice to panic there but it seems like there's no other error reporting at all implemented right now...

Reply:
right, that's why I just gave up there 😅